platform-dom/src/main/java/com/softicar/platform/dom/elements/input/auto/DomAutoCompletePopup.java

```diff
@@ -0,0 +1,191 @@
+package com.softicar.platform.dom.elements.input.auto;
+
+import com.softicar.platform.common.core.i18n.IDisplayString;
+import com.softicar.platform.dom.DomCssClasses;
+import com.softicar.platform.dom.DomI18n;
+import com.softicar.platform.dom.elements.DomDiv;
+import com.softicar.platform.dom.elements.DomElementsCssClasses;
+import com.softicar.platform.dom.elements.DomSpan;
+import com.softicar.platform.dom.event.IDomClickEventHandler;
+import com.softicar.platform.dom.event.IDomEvent;
+import com.softicar.platform.dom.input.auto.DomAutoCompleteList;
+import java.util.ArrayList;
+import java.util.List;
+
+class DomAutoCompletePopup<T> extends DomDiv {
+
+	private final DomAutoCompleteInput<T> input;
+	private final IDomAutoCompleteInputEngine<T> inputEngine;
+	private final List<ValueDisplay> valueDisplays;
+	private int selectionIndex;
+	private String pattern;
+
+	public DomAutoCompletePopup(DomAutoCompleteInput<T> input) {
+
+		this.input = input;
+		this.inputEngine = input.inputEngine;
+		this.valueDisplays = new ArrayList<>();
+		this.selectionIndex = -1;
+		this.pattern = "";
+
+		addCssClass(DomElementsCssClasses.DOM_AUTO_COMPLETE_POPUP);
+	}
+
+	public void refresh() {
+
+		clear();
+		updatePattern();
+
+		var values = inputEngine.findMatches(pattern, DomAutoCompleteList.MAXIMUM_ELEMENT_TO_DISPLAY + 1);
```

Comment:
Unrelated but: I think we should rename `MAXIMUM_ELEMENT_TO_DISPLAY` to `MAXIMUM_ELEMENTS_TO_DISPLAY`, while we're at it.

Reply:
renamed